torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -3175,7 +3187,12 @@ def _post_backward_hook(self, param: Parameter, *unused: Any) -> None:
                 and self._param_exec_order_state == ParamExecOrderState.UNINITIALIZED
             ):
                 # In self._handles_exec_order, the handles are ordered based on
-                # the execution order in the backward pass in the first iteration.
+                # the gradient ready order in the backward pass in the first
+                # iteration.
+                # Note: the gradient ready order of `FlatParameter` may not be
+                # consistent with the order of `nn.Parameter` in the original
```

Comment:
Maybe (to clarify it is not referring to the order in which the parameters are registered/defined):
```suggestion
                # consistent with the execution order of `nn.Parameter` in the original
```